xl-objects.go

```diff
@@ -150,6 +156,8 @@ func (xl xlObjects) GetObject(bucket, object string, startOffset int64) (io.Read
 	if !IsValidObjectName(object) {
 		return nil, ObjectNameInvalid{Bucket: bucket, Object: object}
 	}
+	nsMutex.RLock(bucket, object)
+	defer nsMutex.RUnlock(bucket, object)
```

Comment:
> Can't let go of the lock because it might introduce a race condition, or I did not understand what you said

I mean if you do 'defer' this code would not have unlocked.  Now before the go-routine there is another RLock() which would cause panic.